src/lib/ngsi/Metadata.cpp

```diff
@@ -259,11 +259,14 @@ std::string Metadata::render(const std::string& indent, bool comma)
 
     if (compoundValueP->isObject())
     {
+      // Note in this case we don't add the "value" key, the toJson()
+      // method does it for toplevel compound (a bit crazy... this deserves a FIXME mark)
+      compoundValueP->renderName = true;
       part = compoundValueP->toJson(true, false);
     }
     else if (compoundValueP->isVector())
     {
-      part = "[" + compoundValueP->toJson(true, false) + "]";
+      part = JSON_STR("value") + ": [" + compoundValueP->toJson(true, false) + "]";
```

Comment:
We have plenty of test but... we have a test covering the metadata compound vector rendering in NGSIv1?

Note that "value" is added *but only in the case* of objects. Have a look to CompoundNode::toJson(), the check for "topmost" is done only in the object case.

I think it is not only ok to hardoce "value" here (for the case of compound vector) because no other is going to add it. Again, CompoundNode::toJson() only add it in the case of object, as far as I have seen).

I think we already have that issue on refactoring: https://github.com/telefonicaid/fiware-orion/issues/1298 (however, the issue currently mention only NGSIv2... we should consider if it worth the time/effort to refactor also all the NGSIv1 render logic).